test/integration/device-state.spec.ts

```diff
@@ -424,7 +425,12 @@ describe('device-state', () => {
 					},
 				},
 			} as TargetState),
-		).to.be.rejected;
+		).to.not.be.rejected;
+		const targetState = await deviceState.getTarget();
+		const app = targetState.local.apps[1234];
+		expect(app).to.have.property('appName').that.equals('superapp');
+		expect(app).to.have.property('commit').that.equals('one');
+		expect(app).to.have.property('isRejected').that.is.true;
```

Comment:
What do you think of adding the test cast `accepts target state with valid contract and resets isRejected to false`?

Reply:
Yes, good idea

Reply:
Done

Reply:
I ended up merging the two test cases to remove duplication